### DISCUSSION TITLE
Can't change default branch from branches page

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#53545] [Repositories]
### Select Topic Area

Bug

### Body

1. Visit any repository you can edit
2. Make sure your browser window is *relatively wide* (smaller windows will not show this button)
3. Click `branches`
    <img width="1244" alt="image" src="https://user-images.githubusercontent.com/2119212/233848187-44b48e00-70b9-4ad6-947b-4e259f0391d7.png">
4. Click the Switch default branch button

This is what I see:
<img width="791" alt="image" src="https://user-images.githubusercontent.com/2119212/233847872-9c0f2cdd-88cb-45f1-9d42-0bffd5ce5c1f.png">

---

This used to work.

**Comment by bytepoets-pla** [comment]
Got the same issue on personal and organisation repositories

**Reply by jsoref** [reply]
Fwiw, you can change the branch by going to the main repository /settings page and clicking the same button there:

<img width="651" alt="image" src="https://user-images.githubusercontent.com/2119212/235249271-b959222a-3b87-4f5d-b870-729f4f6c88c1.png">

It's documented here:
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch

But that doesn't justify having a broken button.

**Reply by bytepoets-pla** [reply]
Ah, nice hint, thank you! Somehow the broken button led me to the assumption this feature **should** be located in the branches section and I never thought of looking in "General" 🤦

**Reply by jsoref** [reply]
I never thought to look there and this used to work.

The button here could easily be updated to link to the main settings page...

**Reply by bytepoets-pla** [reply]
Yes, redirection to the "General" Settings page would probably be the behavior I'd expect
